streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -448,6 +448,11 @@ public void shouldNotAllowNullProcessorNameWhenConnectingProcessorAndStateStores
         builder.connectProcessorAndStateStores(null, "store");
     }
 
+    @Test(expected = NullPointerException.class)
+    public void shouldNotAllowNullStateStoreNameWhenConnectingProcessorAndStateStores() throws Exception {
+        builder.connectProcessorAndStateStores("processor", new String[]{null});
```

Comment:
simplify to `.connectProcessorAndStateStores("processor", null);` ?

Reply:
I did this intentionally so that the array itself is not null but the element is.

Reply:
From my understanding, `null` will result in an array with a single `null` element. You would need to create a variable `String[] x = null` and pass `x` to get a `null` array IMHO (I even think, that you cannot do `method("processor", (String[]) null)` as for single argument it would expect a `String` type and thus the cast ends up with incompatible types.) Correct me if I am wrong.

Reply:
From my testing if you just passed `null` it will result in the array as `null` itself directly. Maybe it is JDK dependent (I'm testing with J8).